modules/setting/setting.go

```diff
@@ -867,6 +867,16 @@ func newLogService() {
 	LogModes = strings.Split(Cfg.Section("log").Key("MODE").MustString("console"), ",")
 	LogConfigs = make([]string, len(LogModes))
 
+	useConsole := false
+	for _, mode := range LogModes {
+		if mode == "console"  {
+			useConsole = true
+		}
+	}
+	if (!useConsole) {
+		log.DelLogger("console")
```

Comment:
With this implementation, we always add the console logger (`setting.go: line 421`), and then sometimes remove it later (this if statement). Would it make more sense to check whether we need the console logger when we initially add it, instead of checking after we've already added it?